apiv6/generate_client.sh

```diff
@@ -1,52 +1,79 @@
+synthetics_spec_filename="synthetics.openapi.json"
+synthetics_spec_url="https://raw.githubusercontent.com/kentik/api-schema-public/master/gen/openapiv2/kentik/synthetics/v202101beta1/synthetics.swagger.json"
 
-function stage() {
-    BOLD_BLUE="\e[1m\e[34m"
-    RESET="\e[0m"
-    msg="$1"
-    
-    echo
-    echo -e "$BOLD_BLUE$msg$RESET"
+function run() {
+    check_prerequisites
+
+    download_openapi_spec
+    generate_cloud_export_client
+    generate_synthetics_client
+    go fmt ./...
 }
 
-function checkPrerequsites() {
+function check_prerequisites() {
     stage "Checking prerequisites"
-    
+
     if ! docker --version > /dev/null 2>&1; then
-        echo "You need to install docker to run the generator"
+        echo "You need to install Docker to run the generator"
         exit 1
     fi
 
     echo "Done"
 }
 
-function generate_golang_client_from_openapi3_spec() {
-    stage "Generating golang client from openapi spec"
+function download_openapi_spec() {
+    wget "$cloud_export_spec_url" --output-document "$cloud_export_spec_filename"
+    wget "$synthetics_spec_url" --output-document "$synthetics_spec_filename"
+}
+
+function generate_cloud_export_client() {
+    cloudexport_package="cloudexport"
+    cloudexport_client_output_dir="kentikapi/cloudexport"
+
+    generate_go_client_from_openapi_spec "$cloud_export_spec_filename" "$cloudexport_package" "$cloudexport_client_output_dir"
+    change_ownership_to_current_user "$cloudexport_client_output_dir"
+    cleanup_non_needed_files "$cloudexport_client_output_dir"
+}
+
+function generate_synthetics_client() {
+    synthetics_package="synthetics"
+    synthetics_client_output_dir="kentikapi/synthetics"
+
+    generate_go_client_from_openapi_spec "$synthetics_spec_filename" "$synthetics_package" "$synthetics_client_output_dir"
+    change_ownership_to_current_user "$synthetics_client_output_dir"
+    cleanup_non_needed_files "$synthetics_client_output_dir"
+}
+
+function generate_go_client_from_openapi_spec() {
+    stage "Generating Go client from OpenAPI specification"
 
     spec_file="$1"
     package="$2"
     output_dir="$3"
 
-    docker run --rm  -v "$(pwd):/local" \
-        openapitools/openapi-generator-cli generate  \
+    docker run --rm -v "$(pwd):/local" \
+        "openapitools/openapi-generator-cli:$openapi_generator_tag" generate \
         -i "/local/$spec_file" \
         -g go \
         --additional-properties=enumClassPrefix=true \
         --package-name "$package" \
         -o "/local/$output_dir"
 
-
     echo "Done"
 }
 
 function change_ownership_to_current_user() {
     dir="$1"
     stage "Changing ownership of $dir to $USER:$USER"
 
-    sudo chown  -R "$USER:$USER"  "$dir" # by default the generated output is in user:group root:root
+    sudo chown -R "$USER:$USER" "$dir" # by default the generated output is in user:group root:root
```

Comment:
Does not work in my environment, because there is no group with the same name as my username. MacOS does not create a separate group for each user. I suggest to just use `chown -R "$USER" ...`.

BTW, This has been failing for me for a while, but it does not seem to have any negative effect. Maybe the chown is even not needed.

Reply:
Modified to not change the group. Changing user is needed so that generated files can be removed without `sudo` and formatted by `gofmt`.